server/app/views/applicant/ApplicantFileUploadRenderer.java

```diff
@@ -108,8 +109,14 @@ public UlTag uploadedFiles(
       ApplicationBaseViewParams params, FileUploadQuestion fileUploadQuestion) {
     UlTag result = ul().attr("aria-label", "Uploaded files");
 
+    JsonArray uploadedFileNames = new JsonArray();
+
     if (fileUploadQuestion.getFileKeyListValue().isPresent()) {
       for (String fileKey : fileUploadQuestion.getFileKeyListValue().get()) {
+        String fileName = FileUploadQuestion.getFileName(fileKey);
+
+        uploadedFileNames.add(fileName);
```

Comment:
question (non-blocking): Are we only worried about de-duping file names that are uploaded at the same time? Does it throw an error if we try to upload files with the same name? Is there no issue if we try to upload a file with the same name as one already stored from a previous session?

Reply:
Unfortunately there isn't much we can reasonably do about this since we call directly to AWS with the file upload, and it doesn't return any error (it just replaces the existing file, which isn't that bad). The only option we have is to catch that on the re-direct from AWS and inform the applicant that it happened. (Which I'm not currently planing to do, but it's definitely an option.)